### ISSUE TITLE
Incorrect velocity selected if correct velocity not explicitly computed

### ISSUE THREAD

**Issue by moorepants** [sympy/sympy#23386]
```python
import sympy as sm
import sympy.physics.mechanics as me

L, w = sm.symbols('L, omega')
q1, q2, q3, u1, u2, u3 = me.dynamicsymbols('q1, q2, q3, u1, u2, u3')
t = me.dynamicsymbols._t

A, B, E = sm.symbols('A, B, E', cls=me.ReferenceFrame)

B.orient_axis(A, w*t, A.y)
E.orient_axis(B, q3, B.z)

B.set_ang_vel(A, w*A.y)
E.set_ang_vel(B, u3*B.z)

O, P1, P2 = sm.symbols('O, P1, P2', cls=me.Point)

P1.set_pos(O, q1*B.x + q2*B.y)
P2.set_pos(P1, L*E.x)

O.set_vel(A, 0)
P1.vel(A)
```

returns:

```
Derivative(q1(t), t)*B.x + Derivative(q2(t), t)*B.y - omega*q1(t)*B.z
```

which is correct.

But if you do:

```python
import sympy as sm
import sympy.physics.mechanics as me

L, w = sm.symbols('L, omega')
q1, q2, q3, u1, u2, u3 = me.dynamicsymbols('q1, q2, q3, u1, u2, u3')
t = me.dynamicsymbols._t

A, B, E = sm.symbols('A, B, E', cls=me.ReferenceFrame)

B.orient_axis(A, w*t, A.y)
E.orient_axis(B, q3, B.z)

B.set_ang_vel(A, w*A.y)
E.set_ang_vel(B, u3*B.z)

O, P1, P2 = sm.symbols('O, P1, P2', cls=me.Point)

P1.set_pos(O, q1*B.x + q2*B.y)
P2.set_pos(P1, L*E.x)

O.set_vel(A, 0)
P1.set_vel(B, u1*B.x + u2*B.y)
P1.vel(A)
```

it returns:

```
u1(t)*B.x + u2(t)*B.y
```

which is incorrect! For some reason `B_v_P1` is selected for `A_v_P1` when they are clearly not the same.

The automatic velocity selection introduced in https://github.com/sympy/sympy/pull/20049 and https://github.com/sympy/sympy/pull/21730 gives incorrect results. It should be fixed quickly or completely reverted, as this may cause lots of downstream incorrect equations of motion.

@TJStienstra this must be the same bug you've been working on. This is a simple example that can be used as a test case.

**Comment by moorepants** [comment]
@sidhu1012 FYI

**Comment by moorepants** [comment]
Master gives the correct result:

```python
Python 3.7.12 | packaged by conda-forge | (default, Oct 26 2021, 06:08:53) 
Type 'copyright', 'credits' or 'license' for more information
IPython 7.32.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: %paste
import sympy as sm
import sympy.physics.mechanics as me

L, w = sm.symbols('L, omega')
q1, q2, q3, u1, u2, u3 = me.dynamicsymbols('q1, q2, q3, u1, u2, u3')
t = me.dynamicsymbols._t

A, B, E = sm.symbols('A, B, E', cls=me.ReferenceFrame)

B.orient_axis(A, w*t, A.y)
E.orient_axis(B, q3, B.z)

B.set_ang_vel(A, w*A.y)
E.set_ang_vel(B, u3*B.z)

O, P1, P2 = sm.symbols('O, P1, P2', cls=me.Point)

P1.set_pos(O, q1*B.x + q2*B.y)
P2.set_pos(P1, L*E.x)

O.set_vel(A, 0)
P1.set_vel(B, u1*B.x + u2*B.y)
P1.vel(A)

## -- End pasted text --
Out[1]: Derivative(q1(t), t)*B.x + Derivative(q2(t), t)*B.y - omega*q1(t)*B.z
```

**Comment by TJStienstra** [comment]
This was the bug from #21730 which was reverted with [this PR](https://github.com/sympy/sympy/issues/23358). What happened here is that the assumption was made that after finding the neighbor `O`, the position is obtained from `O` to `P1`. And because there is a `B` frame component in this position vector, it just (incorrectly) assumed that it could just take the sum `A_v_O + B_v_P1`. But as you notice here this issue has been fixed in the master.

**Comment by moorepants** [comment]
Fixed with #21730.
